File: daily_challenge_leetcode/2048.cpp
//Approach-1 (Simple Simulation)
//T.C : O(L-n), where L = 1224444
//S.C : O(1)
class Solution {
public:
    bool balanced(int num) {
        vector<int> freq(10);
        while(num > 0) {
            int digit = num%10;
            freq[digit]++;
            num /= 10;
        }
        for(int d = 0; d < 10; d++) {
            if(freq[d] != 0 && freq[d] != d)
                return false;
        }
        return true;
    }
    int nextBeautifulNumber(int n) {
        for(int num = n+1; num <= 1224444; num++) {
            if(balanced(num)) {
                return num;
            }
        }
        return -1;
    }
};

File: daily_challenge_leetcode/2048.js
/**
 * @param {number} n
 * @return {number}
 */
var nextBeautifulNumber = function (n) {
    const balanced = (num) => {
        const freq = new Array(10).fill(0)
        while (num > 0) {
            const digit = num % 10
            freq[digit]++
            num = Math.floor(num / 10)
        }
        for (let d = 0; d < 10; d++) {
            if (freq[d] !== 0 && freq[d] !== d) {
                return false
            }
        }
        return true
    }
    for (let num = n + 1; num <= 1224444; num++) {
        if (balanced(num)) {
            return num;
        }
    }
    return -1;
};

